### ISSUE TITLE
Reintroduce satellite point click

### ISSUE THREAD

**Issue by ajmas** [ajmas/StuffInSpace#20]
Reintroduce the ability to click on a satellite to get information on it.

**Comment by ajmas** [comment]
Re-added, but need to work on the raycaster logic, since we end up grabbing the wrong satellite in a good number of cases

**Comment by thkruz** [comment]
The original implementation uses GPU picking not raycasting (much faster). The gist of it is that you create different colors for every dot and then write those to a buffer that isn't rendered to the canvas. Each draw call there is a second call to the GPU for the pixel color on that buffer under the mouse.

I added a throttle on my implementation to only run that every 8 or 16 frames to reduce latency without really even noticing.

The more important thing is using an asyncReadPixel call (can't remember if that is WebGl 2.0 only). It might lead you to being slightly behind the mouse, but again it is a non-issue compared to blocking the main thread waiting on the GPU.

Are you set on raycasting for another reason or are you open to sticking with GPU picking?

**Comment by ajmas** [comment]
I am open to any approach that works and if there is a better solution, then I'd make use of that.

My current choice of raycasting was more that I am still a newbie in 3D and was focusing on function. Optimisation wasn't an immediate priority and my plan was to go back and compare the old behaviour

For the asyncReadPixel I'll need to read up on that. BTW this is the same as `gl.readPixels`?

**Comment by ajmas** [comment]
I also opened issue #34 on targeting of lines. Will see if this can be addressed as part of this work or whether it needs to stay its own ticket.

**Comment by ajmas** [comment]
Since there is currently a working solution using the ThreeJS reaycaster, I have moved the suggestion of using the asyncReadPixel to issue #45.
